### DISCUSSION TITLE
Non-root site path not working

### DISCUSSION THREAD

**Discussion by michielvdmeulen** [quarto-dev/quarto-cli#3615] [Q&A]
I'm trying to serve a blog on a non-root path, but despite following the instructions in the [documentation](https://quarto.org/docs/websites/website-navigation.html#non-root-site-paths) I can't get it to work. My situation is identical to that in the docs, apart from different url naming of course, so something like this:

URL:
https://example.com/mysite/ 

_quarto.yml:
```
website:
  title: "My Site"
  site-path: "/mysite/"
```

When I render the site with and without the `site-path` option and compare the generated `index.html` file in the `_site` folder, both are exactly identical - even when I have completely deleted and re-rendered the content. In fact, only the `_site\site_libs\quarto-nav\quarto-nav.js` file has one line difference:
![Screenshot_1](https://user-images.githubusercontent.com/15234562/206665447-fe8aa527-bf25-4b13-a1a4-d0ea40750b3b.png)
Apparently, this is either not enough or is not picked up on the server side. I'm using the latest Quarto version (1.2.269).

Does someone have an idea what could be wrong? Any help would be highly appreciated!

**Comment by cscheid** [comment]
The content should in fact be almost identical: the way we make non-root site paths work is that we convert every absolute path to the correct relative path on the web page.

Can you give specifics for what's going wrong with the rendered website?

**Reply by michielvdmeulen** [reply]
Thanks for the quick reply. I can only say that the website, when rendered locally on localhost, does not get `/mysite/` appended to the URL (I don't know if that is the expected behaviour). On the server side, it is hard for me to say what is not working, because the site is hosted on an Azure server that I cannot inspect or setup myself since I don't have permissions - a colleague of mine is doing that for me. He mentioned he would expect the `index.html` file to contain a modification to the URL to direct to `/mysite/`. 

The server runs a docker container by the way, which uses `nginx` to serve the site. At this point, I'm not using any interactive content or computation engine on the server, it is just supposed to show the rendered content in the `_site` folder.

**Reply by dragonstyle** [reply]
You generally don't need to provide a `site-path` (this really only affects 404 pages afaik). You can more generally provide a `site-url` like so:

```
website:
  favicon: logo.png
  site-url: https://example.com/mysite/
```

And this will ensure that anywhere Quarto is using an absolute path, we use the proper prefix. In most places, we use relative paths, so it is expected that the `mysite/` doesn't appear in urls.

**Reply by TalalMahmoodChaudhry** [reply]
Hi,

If I add the site-url (or site-path) and then use `quarto preview` to preview the website. It still opens at `Browse at http://localhost:7863/`.  The same result if I render it and then run it. Why doesn't it append `/mysite/` to it. So how can I make it so it will be like `http://localhost:7863/myside/`. 

Our goal is to run the website in Kubernetes. In the ingress file of kubernetes, I have set it up so that all traffic that comes tp `/mysite/` should route traffic to the pod running the website. And this is not working because I am unable to run the website with a root path of `/mysite/`.

**Reply by TalalMahmoodChaudhry** [reply]
I got it working. Just changed the kubernetes ingress route traffic coming to `/mysite/` to `/` of the pod. Thank you

**Comment by mehalter** [comment]
I am also running into this issue where deploying to a non-root url just doesn't work. It appears that most things use relative domains, but not everywhere. Here is a small snippet from my rendered `index.html`:

```html
<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./_assets/brand/favicon.ico" rel="icon">
```

here we can see that the link to the `favicon.ico` is set up appropriately, but non of the core `site_lib` references use relative domains. This causes none of the javascript/css to work appropriately when the website is not hosted on a root domain

**Reply by mehalter** [reply]
Oh, I see, this causes a weirdness. So when I go my website on the subdomain without a trailing `/` the site doesn't work. So for example I have this deployed to https://www.algebraicjulia.org/blog/ but if you go to https://www.algebraicjulia.org/blog the website does not function correctly.

**Reply by cscheid** [reply]
@mehalter That's a "fun" one, thanks for catching. Can you open a new issue? We definitely want to fix this.

**Reply by mehalter** [reply]
Sure thing! I will open one later today when I get some time to sit down and make a well formulated issue and include relevant details. Wanted to see if I was being silly initially before making a bunch of noise in the issue list :)

**Reply by cscheid** [reply]
> Wanted to see if I was being silly initially before making a bunch of noise in the issue list

We deeply appreciate the mindfulness.

**Reply by reuning** [reply]
@mehalter did you open an issue on this? I realized I'm having the same problem. It isn't a big deal but it is strange.

**Comment by nvelden** [comment]
I have the same issue as described but with a book document. It is described here: https://stackoverflow.com/questions/74463751/how-can-i-explicitly-link-to-the-index-html-file-in-a-quarto-book. Using the suggested solution of setting `title: "<a href='/index.html'>TGP User Guide</a>` doesn't work for me.
